transport/graphsync/graphsync.go

```diff
@@ -709,15 +724,18 @@ func (t *Transport) gsRequestorCancelledListener(p peer.ID, request graphsync.Re
 	}
 }
 
-func (t *Transport) gsNetworkErrorListener(p peer.ID, request graphsync.RequestData, err error) {
+// Called when there is a graphsync error sending data
+func (t *Transport) gsNetworkSendErrorListener(p peer.ID, request graphsync.RequestData, gserr error) {
 	t.dataLock.Lock()
 	defer t.dataLock.Unlock()
 
 	chid, ok := t.graphsyncRequestMap[graphsyncKey{request.ID(), p}]
-	if ok {
```

Comment:
Firing a Disconnect event here would cause go-fil-markets to fail the deal.
Instead we fire a SendDataError which the push / pull monitor listen for - they will attempt to reconnect and restart the channel